CONTRIBUTING.md

```diff
@@ -0,0 +1,61 @@
+
+!!! note "Shortcut"
+    If you are looking through the docs and find an issue, hit the :material-file-edit: button on the top right corner of the page. This will take you to the underlying file on the Splink GitHub page.
+    
+    This is not supported for Tutorials or Examples as they are in a separate repo.
+
+For small changes, such as wording and typos, changes can be made directly in GitHub. However, for larger changes it may be worth cloning the relevant repo to your local machine. This way, you can [build the docs site locally](./docs/dev_guides/changing_splink/build_docs_locally.md) to check how the changes will look in the deployed doc site.
+
+
+## Contributing code
+Thanks for your interest in contributing code to Splink!
+
+There are a number of ways to get involved:
+
+- Start work on an [existing issue](https://github.com/moj-analytical-services/splink/issues), there should be some with a [`good first issue`](https://github.com/moj-analytical-services/splink/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) flag which are a good place to start. 
+- Tackle a problem you have identified. If you have identified a feature or bug, the first step is to [create a new issue](https://github.com/moj-analytical-services/splink/issues/new/choose) to explain what you have identified and what you plan to implement, then you are free to fork the repo and get coding!
+
+In either case, we ask that you assign yourself to the relevant issue and open up [a draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) while you are working on your feature/bug-fix. This helps the Splink dev team keep track of developments and means we can start supporting you sooner!
+
+We do not work with a `dev` branch, so any branches should be made from `master`.
+
+!!! note "Small PRs"
+    In the Splink dev team, we believe that [small Pull Requests](https://essenceofcode.com/2019/10/29/the-art-of-small-pull-requests/) make better code. They:
+
+    - are more focused
+    - increase understanding and clarity
+    - are easier (and quicker) to review
+    - get feedback quicker
+
+    You can always add further PRs to build extra functionality. Starting out with a minimum viable product and iterating  makes for better software (in our opinion). It also helps get features out into the wild sooner with regular Splink releases every other Wednesday.
+
+When making code changes, we recommend:
+* [Recreating Splink's virtual environment](./docs/dev_guides/changing_splink/building_env_locally.md) to best replicate the conditions in which Splink will be used in practice. 
+* [Adding tests](./docs/dev_guides/changing_splink/testing.md) to ensure your code works as expected. These will be run through GitHub Actions when a PR is opened.
+* [Linting](./docs/dev_guides/changing_splink/lint_and_format.md) to ensure that code is styled consistently.
```

Comment:
Do we want a note on CI tests running on the creation of a PR and how we expect these to pass?

Historically we've also had issues here as some tests fail due to the code existing on two different repositories - see an example [here](https://github.com/moj-analytical-services/splink/pull/1114).

I believe these have all been resolved now, but it might be worthwhile adding a caveat if you add something in about CI.

Reply:
Slight side-note, but if adding something about CI it might be worth highlighting that the 'benchmark-comment' workflow will currently fail from forks (I think I was fixing that at some point??), but the rest are expected to pass before merge